dotCMS/src/main/java/com/dotcms/rest/config/DotRestApplication.java

```diff
@@ -67,111 +69,129 @@
+                    .add(ConditionResource.class)
+                    .add(ConditionValueResource.class)
+                    .add(PersonasResourcePortlet.class)
+                    .add(ConditionletsResource.class)
+                    .add(MonitorResource.class)
+                    .add(ActionResource.class)
+                    .add(ActionletsResource.class)
+                    .add(I18NResource.class)
+                    .add(LanguagesResource.class)
+                    .add(com.dotcms.rest.api.v2.languages.LanguagesResource.class)
+                    .add(MenuResource.class)
+                    .add(AuthenticationResource.class)
+                    .add(LogoutResource.class)
+                    .add(LoginFormResource.class)
+                    .add(ForgotPasswordResource.class)
+                    .add(ConfigurationResource.class)
+                    .add(AppContextInitResource.class)
+                    .add(SiteResource.class)
+                    .add(ContentTypeResource.class)
+                    .add(FieldResource.class)
+                    .add(com.dotcms.rest.api.v2.contenttype.FieldResource.class)
+                    .add(com.dotcms.rest.api.v3.contenttype.FieldResource.class)
+                    .add(FieldTypeResource.class)
+                    .add(FieldVariableResource.class)
+                    .add(ResetPasswordResource.class)
+                    .add(RoleResource.class)
+                    .add(CreateJsonWebTokenResource.class)
+                    .add(ApiTokenResource.class)
+                    .add(PortletResource.class)
+                    .add(EventsResource.class)
+                    .add(FolderResource.class)
+                    .add(BrowserTreeResource.class)
+                    .add(CategoriesResource.class)
+                    .add(PageResource.class)
+                    .add(ContentRelationshipsResource.class)
+                    .add(com.dotcms.rest.api.v1.workflow.WorkflowResource.class)
+                    .add(ContainerResource.class)
+                    .add(ThemeResource.class)
+                    .add(NavResource.class)
+                    .add(RelationshipsResource.class)
+                    .add(VTLResource.class)
+                    .add(ContentVersionResource.class)
+                    .add(FileAssetsResource.class)
+                    .add(PersonalizationResource.class)
+                    .add(TempFileResource.class)
+                    .add(UpgradeTaskResource.class)
+                    .add(AppsResource.class)
+                    .add(BrowserResource.class)
+                    .add(ResourceLinkResource.class)
+                    .build();
+
+	
+	/**
+	 * This is the cheap way to create a concurrent set of user provided classes
+	 */
+    private final static Map<Class<?>, Boolean> customClasses = new ConcurrentHashMap<>();
+
+    /**
+     * adds a class and reloads
+     * @param clazz
+     */
+    public synchronized static void addClass(Class<?> clazz) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use block level rather than method level synchronization](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5566474)